core/src/main/java/io/kestra/core/runners/Worker.java

```diff
@@ -291,7 +291,7 @@ private void handleTask(WorkerTask workerTask) {
                     );
 
                     // all tasks will be handled immediately by the worker
-                    WorkerTaskResult workerTaskResult = this.run(currentWorkerTask, false);
+                    WorkerTaskResult workerTaskResult = this.run(currentWorkerTask.withRunContext(runContext.forWorkingDirectoryTask()), false);
```

Comment:
I don't think you need to add the `@With` on the `WorkerTask`. Maybe you can just pass the runContext returned from the `forWorkingDirectoryTask` method directly when initializing the `currentWorkerTask` variable on line 287.